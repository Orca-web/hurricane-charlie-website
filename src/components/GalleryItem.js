import React from 'react';

const GalleryItem = () => {
  return <div className="grid-item">Item</div>;
}

export default GalleryItem;