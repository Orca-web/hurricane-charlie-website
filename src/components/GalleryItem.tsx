import React from 'react';
import { IGalleryItem } from '../globals/galleryData';

interface GalleryItemProps {
  galleryItemData: IGalleryItem;
  selectGalleryItem: (id: number) => void;
  imgData: string;
}

const GalleryItem = ({
  galleryItemData,
  selectGalleryItem,
  imgData,
}: GalleryItemProps): JSX.Element => {
  const itemClass = `grid-item ${galleryItemData ? galleryItemData.size : null}`;
  
  return (
    <div
      onClick={() => selectGalleryItem(galleryItemData.id)}
      className={itemClass}
      style={{ backgroundImage: `url('${imgData}')`}}
    >
      <div className="img-overlay"><p>{galleryItemData.name}</p></div>
    </div>
  );
}

export default GalleryItem;