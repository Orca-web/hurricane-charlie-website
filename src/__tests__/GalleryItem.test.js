import React from 'react';
import { shallow } from 'enzyme';
import GalleryItem from '../components/GalleryItem';

const app = shallow(<GalleryItem />);

describe("Gallery Item", () => {
  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });
});