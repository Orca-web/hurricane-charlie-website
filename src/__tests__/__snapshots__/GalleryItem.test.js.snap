// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Item renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GalleryItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Item",
      "className": "grid-item",
    },
    "ref": null,
    "rendered": "Item",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Item",
        "className": "grid-item",
      },
      "ref": null,
      "rendered": "Item",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
